File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def reverse(self, x: int) -> int:
        
        MAX,MIN=2147483647,-2147483648
        res=0
        while x:
            digit=math.fmod(x,10)#-1%10=9 in py 

            x=int(x/10)# -1//10=-1

            if res>int(MAX/10) or (res==MAX and digit>7):
                return 0
            if res<int(MIN/10) or (res==MIN and digit<8):
                return 0
            res=(res*10)+digit
        return int(res)

File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l=0,r=heights.length-1,max_area=0
        while(l<r){
            let curr_h=Math.min(heights[l],heights[r])
            let curr_w=r-l
            let curr_area=curr_h*curr_w
            if(heights[l]<heights[r]) l+=1
            else r-=1
            max_area=Math.max(max_area,curr_area)

        }
        return max_area
    }
}
